--- a/draminit/dwc/Makefile	2024-10-21 00:16:39.272937518 +0300
+++ a/draminit/dwc/Makefile	2024-10-25 18:18:03.267314631 +0300
@@ -18,56 +18,62 @@
 endif
 
 PROJECT_ROOT = $(shell cd ../../xboot/ && pwd)
-SPI_NAND_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_SPI_NAND")
+ifneq ($(O),)
+PROJECT_ROOT = $(O)../../
+endif
+SPI_NAND_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_SPI_NAND")
 ifeq (${SPI_NAND_CFG},CONFIG_HAVE_SPI_NAND=y)
 CFLAGS += -DCONFIG_HAVE_SPI_NAND
 endif
-HAVE_SNPS_USB3_DISK_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_SNPS_USB3_DISK")
+HAVE_SNPS_USB3_DISK_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_SNPS_USB3_DISK")
 ifeq (${HAVE_SNPS_USB3_DISK_CFG},CONFIG_HAVE_SNPS_USB3_DISK=y)
 CFLAGS += -DCONFIG_HAVE_SNPS_USB3_DISK
 endif
-HAVE_USB_DISK_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_USB_DISK")
+HAVE_USB_DISK_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_USB_DISK")
 ifeq (${HAVE_USB_DISK_CFG},CONFIG_HAVE_USB_DISK=y)
 CFLAGS += -DCONFIG_HAVE_USB_DISK
 endif
-HAVE_PARA_NAND_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_PARA_NAND")
+HAVE_PARA_NAND_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_HAVE_PARA_NAND")
 ifeq (${HAVE_PARA_NAND_CFG},CONFIG_HAVE_PARA_NAND=y)
 CFLAGS += -DCONFIG_HAVE_PARA_NAND
 endif
 
 # Get Platform configuration from xboot/.config
 PROJECT_ROOT = $(shell cd ../../../boot/xboot && pwd)
+ifneq ($(O),)
+PROJECT_ROOT = $(O)../../
+endif
 # Get Dram type
-HAVE_LPDDR4_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_LPDDR4")
+HAVE_LPDDR4_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_LPDDR4")
 ifeq (${HAVE_LPDDR4_CFG},CONFIG_LPDDR4=y)
 CFLAGS += -DCONFIG_LPDDR4
 endif
-HAVE_DDR4_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_DDR4")
+HAVE_DDR4_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_DDR4")
 ifeq (${HAVE_DDR4_CFG},CONFIG_DDR4=y)
 CFLAGS += -DCONFIG_DDR4
 endif
-HAVE_LPDDR3_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_LPDDR3")
+HAVE_LPDDR3_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_LPDDR3")
 ifeq (${HAVE_LPDDR3_CFG},CONFIG_LPDDR3=y)
 CFLAGS += -DCONFIG_LPDDR3
 endif
-HAVE_DDR3_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_DDR3")
+HAVE_DDR3_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_DDR3")
 ifeq (${HAVE_DDR3_CFG},CONFIG_DDR3=y)
 CFLAGS += -DCONFIG_DDR3
 endif
 # Get Dram part number
-HAVE_NT6AN1024F32AV_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_NT6AN1024F32AV")
+HAVE_NT6AN1024F32AV_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_NT6AN1024F32AV")
 ifeq (${HAVE_NT6AN1024F32AV_CFG},CONFIG_NT6AN1024F32AV=y)
 CFLAGS += -DCONFIG_NT6AN1024F32AV
 endif
-HAVE_MT53D1024M32D4_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53D1024M32D4")
+HAVE_MT53D1024M32D4_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53D1024M32D4")
 ifeq (${HAVE_MT53D1024M32D4_CFG},CONFIG_MT53D1024M32D4=y)
 CFLAGS += -DCONFIG_MT53D1024M32D4
 endif
-HAVE_MT53E1G32D2_A_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53E1G32D2_A")
+HAVE_MT53E1G32D2_A_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53E1G32D2_A")
 ifeq (${HAVE_MT53E1G32D2_A_CFG},CONFIG_MT53E1G32D2_A=y)
 CFLAGS += -DCONFIG_MT53E1G32D2_A
 endif
-HAVE_MT53E1G32D2_B_CFG := $(shell cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53E1G32D2_B")
+HAVE_MT53E1G32D2_B_CFG := $$(cat $(PROJECT_ROOT)/.config | grep "CONFIG_MT53E1G32D2_B")
 ifeq (${HAVE_MT53E1G32D2_B_CFG},CONFIG_MT53E1G32D2_B=y)
 CFLAGS += -DCONFIG_MT53E1G32D2_B
 endif
@@ -149,21 +155,22 @@
 CSOURCES += $(DWC_USER_DIR)/$(DWC_USER)io_write16.c
 CSOURCES += $(DWC_USER_DIR)/$(DWC_USER)io_read16.c
 CSOURCES += $(DWC_USER_DIR)/$(DWC_USER)saveRetRegs.c
-OBJS = $(CSOURCES:.c=.o)
+OBJS = $(addprefix $(O),$(CSOURCES:.c=.o))
 
 
 all: $(OBJS)
 
-%.o: %.c
+$(O)%.o: %.c
+	install -d $(dir $@)
 	$(CC) $(CFLAGS) -c -o $@ $<
 
 # Dependency
-.depend: $(CSOURCES)
-	@rm -f .depend >/dev/null
-	@$(CC) $(CFLAGS) -MM $^ >> ./.depend
-sinclude .depend
+$(O)dwc.depend: $(CSOURCES)
+	@rm -f $(O)dwc.depend >/dev/null
+	@$(CC) $(CFLAGS) -MM $^ >> $(O)dwc.depend
+sinclude $(O)dwc.depend
 
 .PHONY: clean
 clean:
 	rm -f $(OBJS)
-	rm -f .depend
+	rm -f $(O).depend
